Updates documentation - Priors

guillermoaguilar opened the pull request:
updates documentation on priors

- explanation default priors
- how to change stimulus range for adaptive procedures
- pass a custom prior function
- check information against matlab wiki

updates also function that plots psychometric function,

- adds a parameter to scale the marker size for the data

pberkes commented on docs/examples/priors.md:
```diff
@@ -64,18 +61,41 @@ res = ps.psignifit(data, sigmoid='norm', experiment_type='2AFC', debug=True)
+For a threshold prior to be adequate (first column), it should be flat along all your stimulus levels; this will ensure that the threshold estimation is driven exclusively by the data. The stimulus levels of your data are shown as black-dots in the x-axis.
+The prior is flat (constant) for all threshold values between the red and blue sigmoids; both are equally likely. The yellow and green sigmoids are very unlikely, and this makes sense as they are outside of the range of stimulus levels. Here you can see that `psignifit` assumes that you choose stimulus values along the whole range the psychometric function.
+
+Similarly, for the width prior (middle column) the sigmoids between varing between the red and blue cases are more or less likely; contrarily the yellow and green sigmoids are very unlikely. Again this make sense, as `psignifit` assumes that you sampled the whole range of the psychometric function and not just around the threshold.
```
a "between" too many

pberkes replied:
this "makes" sense

pberkes replied:
I fixed these two typos myself

pberkes commented:
@otizonaizit  this page shows that in some real-life cases it makes sense to adjust the stimulus range

pberkes commented:
LGTM

otizonaizit commented:
> @otizonaizit this page shows that in some real-life cases it makes sense to adjust the stimulus range

Hmm, and yet I fail to make sense of that example. I am not enough into the matter to judge what it means that with an "adaptive" procedure where you don't test stimuli below threshold, by setting the range of the prior you magically get more info about the region you didn't test... I really think this is a discussion to be held again with the experts. @FelixWichmann : what do you think of this example? Is it sound?
If this thing makes sense, then `stimulus_range` must be public.

guillermoaguilar commented:
> > @otizonaizit this page shows that in some real-life cases it makes sense to adjust the stimulus range
> 
> Hmm, and yet I fail to make sense of that example. I am not enough into the matter to judge what it means that with an "adaptive" procedure where you don't test stimuli below threshold, by setting the range of the prior you magically get more info about the region you didn't test... I really think this is a discussion to be held again with the experts. @FelixWichmann : what do you think of this example? Is it sound? If this thing makes sense, then `stimulus_range` must be public.

This documentation page is just a mirror copy of the one for the MATLAB version 

https://github.com/wichmann-lab/psignifit/wiki/Priors

I just adapted the syntax and added some more text to explain the graphs. So I think it is good as it is.

And FYI, actually adaptive procedures are very commonly used, instead of testing stimulus levels in the whole range. People do this to save time, but at a cost. After they collect the data they still want to fit a psychometric function. The problem is that usually the data does not cover the whole function, it tends to be on the higher side, simply  because this is how the adaptive procedure works: roughly speaking, it presents you a high stimulus value, if you correctly answer after X times it lowers the value, and repeat this until you make mistakes... then the procedure makes task easier by increasing the stimulus value. After X number of 'reversals' it stops. 
Then with such a procedure is very likely that you don't sample the whole range of the psychometric function.

I'd say then we will need the `stimulus_range` to be public... this wasn't clear to me before when we removed it :/
